crates/api_crud/src/user/create.rs

```diff
@@ -472,28 +453,35 @@ fn get_language_tags(req: &HttpRequest) -> Vec<String> {
 async fn create_local_user(
   context: &Data<LemmyContext>,
   language_tags: Vec<String>,
-  local_user_form: &LocalUserInsertForm,
-  local_site_id: SiteId,
+  mut local_user_form: LocalUserInsertForm,
+  local_site: &LocalSite,
 ) -> Result<LocalUser, LemmyError> {
   let all_languages = Language::read_all(&mut context.pool()).await?;
   // use hashset to avoid duplicates
   let mut language_ids = HashSet::new();
 
   // Enable languages from `Accept-Language` header
-  for l in language_tags {
-    if let Some(found) = all_languages.iter().find(|all| all.code == l) {
+  for l in &language_tags {
+    if let Some(found) = all_languages.iter().find(|all| &all.code == l) {
       language_ids.insert(found.id);
     }
   }
 
   // Enable site languages. Ignored if all languages are enabled.
-  let discussion_languages = SiteLanguage::read(&mut context.pool(), local_site_id).await?;
+  let discussion_languages = SiteLanguage::read(&mut context.pool(), local_site.site_id).await?;
   language_ids.extend(discussion_languages);
 
   let language_ids = language_ids.into_iter().collect();
 
+  local_user_form.default_listing_type = Some(local_site.default_post_listing_type);
+  local_user_form.post_listing_mode = Some(local_site.default_post_listing_mode);
+  // If its the initial site setup, they are an admin
+  local_user_form.admin = Some(!local_site.site_setup);
+  local_user_form.interface_language = language_tags.first().cloned();
+  // Initialize donation time, this way dialog will first be shown 4 weeks after signup.
+  local_user_form.last_donation_notification = Some(Utc::now() - TimeDelta::weeks(48));
```

Comment:
This one isn't necessary, you can just use `None`, and postgres will handle the default correctly.